### DISCUSSION TITLE
'grunt create-sim' fails on Windows with "Error: spawn EINVAL"

### DISCUSSION THREAD

**Discussion by fgamador** [scenerystack/community#48] [SceneryStack]
When using a recent version of Node on Windows, 'grunt create-sim' fails with "Error: spawn EINVAL". I found that this is a [known issue](https://github.com/nodejs/node/issues/52554) with recent versions of Node and can be fixed by adding `shell: true` to the options passed to `child_process.spawn` in perennial/js/common/execute.js (line 52). I tried it, and it works. I can submit a pull request if you'd like.

C:\Users\franz\Dev\NodeProjects\phetsims\perennial (main -> origin) (perennial@0.0.0)
λ grunt create-sim --repo=my-first-sim --author="Franz Amador" 
Running "create-sim" task
Greetings Franz Amador!
creating sim with repository name my-first-sim
wrote ../my-first-sim/.gitignore
[...]
wrote ../my-first-sim/tsconfig.json
info: npm update in ../my-first-sim
Fatal error: Perennial task failed:
Error: spawn EINVAL
    at ChildProcess.spawn (node:internal/child_process:421:11)
    at Object.spawn (node:child_process:761:9)
    at C:\Users\franz\Dev\NodeProjects\phetsims\perennial\js\common\execute.js:52:40
    at new Promise (<anonymous>)
    at module.exports (C:\Users\franz\Dev\NodeProjects\phetsims\perennial\js\common\execute.js:45:10)
    at C:\Users\franz\Dev\NodeProjects\phetsims\perennial\js\common\npmUpdateDirectory.js:28:11
    at C:\Users\franz\Dev\NodeProjects\phetsims\perennial\node_modules\async-mutex\lib\Mutex.js:23:66
    at Semaphore.<anonymous> (C:\Users\franz\Dev\NodeProjects\phetsims\perennial\node_modules\async-mutex\lib\Semaphore.js:37:46)
    at step (C:\Users\franz\Dev\NodeProjects\phetsims\perennial\node_modules\tslib\tslib.js:195:27)
    at Object.next (C:\Users\franz\Dev\NodeProjects\phetsims\perennial\node_modules\tslib\tslib.js:176:57)
Full Error details:
Error: spawn EINVAL

**Comment by fgamador** [comment]
I added Perennial issue [#359](https://github.com/phetsims/perennial/issues/359) for this.

**Comment by brettfiedler** [comment]
Thanks for catching this and proposing a fix, @fgamador! This was a wonderful contribution. Looks like it touched a few other places in the code and @zepumph needed to track it down. 

Would you be willing to share what brought you to SceneryStack? Feel free to message the chat or DM me (same name) on [Slack/Matrix ](https://scenerystack.github.io/community/join/) as well if you prefer.

**Reply by fgamador** [reply]
You're very welcome. Glad to be able to help, and thanks for being so responsive.

I'm a recently retired software developer ([LinkedIn](www.linkedin.com/in/franz-amador-66b9152a)) looking around for ways to keep my hand in. I've long been interested in biology-related simulation ([GitHub](https://github.com/fgamador?tab=repositories)), and I ran across PhET Sims when I was investigating what's out there in biology-education software. I have some ideas for biology-ed sims that I'd like to try out, and SceneryStack looks like an excellent platform to build on.

**Reply by brettfiedler** [reply]
Thanks @fgamador! I brought the conversation over to Slack.
